sys/filesys_windows.go

```diff
@@ -257,12 +260,71 @@ func windowsOpenSequential(path string, mode int, _ uint32) (fd windows.Handle,
 	return h, e
 }
 
-// ForceRemoveAll is the same as os.RemoveAll, but uses hcsshim.DestroyLayer in order
-// to delete container layers.
+// ForceRemoveAll is the same as os.RemoveAll, but is aware of io.containerd.snapshotter.v1.windows
```

Comment:
These layers may not have been mounted by containers, they may have been mounted to the host OS, where there is no such guarantee. Although I don't think containerd does this _now_, #4419 reproduces this issue much more reliably in the Snapshot Testsuite, where there are no containers, only ~~Zuul~~ host-mounted WCOW layers.

That said, I haven't tested that this is necessary. I do believe `ForceRemoveAll` would be expected to handle the "Things were not cleaned up properly" cases. Otherwise the `RemoveLayer` call fails, and you get bonus failures from that *on top of* the failure that left you half-cleaned-up.

_And_ you have half-mounted layers stuck around until next reboot, which has happened to me with the containerd test suite in the past.

Reply:
That makes sense.